content/2017.02.handling-python-projects-dependencies.rst

```diff
@@ -0,0 +1,225 @@
+Handling Python Project dependencies
+####################################
+
+:slug: handling-python-projects-dependencies
+:date: 2017-02-24
+:lang: en
+:url: handling-python-projects-dependencies
+:summary:
+    Coping with project depdendencies can be a bit cumbersome.
+	Learn how we are handling our Python project dependencies.
+
+
+Introduction
+============
+
+If you ever started a Python project you might have heard of a
+``requirements.txt`` file that contains a list of dependencies.
+
+.. code-block:: ini
+
+    pelican
+    beautifulsoup4
+    ghp-import
+    Pillow==4.0.0
+    markdown
+    pysvg>=0.2,<0.3
+
+
+If you ship you project or library in Pypi you need to put them in the
+``setup.py`` file in order for them to be installed with the other
+projects.
+
+.. code-block:: ini
+
+    setup(name='saucisson',
+      version='1.0.0.dev0',
+      license='Apache License (2.0)',
+      packages=find_packages(),
+      include_package_data=True,
+      zip_safe=False,
+      install_requires=[
+        'colander >= 1.3.2',
+        'colorama',
+        'cornice >= 2.4',
+        'cornice_swagger >= 0.5',
+        'jsonschema',
+        'jsonpatch',
+        'python-dateutil',
+        'pyramid > 1.8',
+        'pyramid_multiauth >= 0.8',  # User on policy selected event.
+        'transaction',
+        'pyramid_tm',
+        'requests',
+        'structlog >= 16.1.0',
+        'waitress',
+        'ujson >= 1.35'])
+
+This brinks a lot of questions:
+
+ - How do I update project dependencies?
+ - What were the know working dependencies version at a certain release?
+ - How do I make sure a release of one of the dependency I use doesn't break my release?
+ - How do I make sure that my code is not used with a non supported version of a dependency?
+
+
+Avoid using requirements file to handle your program dependencies
+=================================================================
+
+Because your program will eventually be released, you want to be able
+to use ``python setup.py install`` to install it so that you will be
+able to install it with ``pip install saucisson`` to install the last
+release and it will install itself automatically with the right
+dependencies.
+
+Instead use your ``requirements.txt`` to set a know working set of your
```

Comment:
Because something is missing in between 🗡 

Because your program will eventually be released, you want to be able
to use ``python setup.py install`` to install it so that you will be
able to install it with ``pip install saucisson`` to install the last
release and it will install itself automatically with the right
dependencies.